src/main/java/adapters/default_game/GamePanelPresenter.java

```diff
@@ -104,6 +69,9 @@ public Graphics2D graphics() {
                 return g2;
             }
         };
+        MazeItems items = new MazeItems();
+        MazeHazards hazards = new MazeHazards();
+        new CustomAssetSetter("mazes/maze02.txt", items, hazards);
```

Comment:
I think this is the problem. Every time paintComponent is called, it's constructing a new MazeItems, MazeHazards, AssetSetter and drawing the map. With this set up we can't reflect changes to the Hazards/Items. The game keeps drawing the original setup.
This is why I think when the Controller & Presenter are initiated, they should take in the same Items & Hazards